### PR TITLE
Gallery: Add link to option

### PR DESCRIPTION
**Pull request by mkaz**
Allow user to select link to option for images similar to core gallery.
Add to Inspector Link To options: Attachment Page, Media File, or None.

See: #1448

### CODE REVIEW

**Review by timmyc — COMMENTED** [review]
While testing this out I hit an issue with linking to the attachment page.  Steps to reproduce:

1- Add new gallery block
2- Select a few images
3- Link To "Attachment Page"
4- Save

When viewing the post on the front-end of the site, the images in the gallery are not linked.  It does work for linking to "Media File"

File: blocks/library/gallery/index.js [hunk 1]
```diff
@@ -136,6 +144,12 @@ registerBlockType( 'core/gallery', {
 						<p>{ __( 'Image galleries are a great way to share groups of pictures on your site.' ) }</p>
 					</BlockDescription>
 					<h3>{ __( 'Gallery Settings' ) }</h3>
+					<SelectControl
+						label={ __( 'Link to' ) }
+						selected={ linkto }
+						onBlur={ setLinkto }
```
Comment (timmyc) [inline blocks/library/gallery/index.js]: Any reason to not have `onChange={ setLinkto }` here?  Currently, after changing the value of this element, you must click off the select drop down to trigger the `onBlur` before you can update the post:

no-on-change

**Comment by timmyc** [comment]
I also re-triggered the CI job that had failed, all looks green now.

File: blocks/library/gallery/index.js (hunk 1 above)
Comment (mkaz) [inline blocks/library/gallery/index.js]: Ah good point, I'll update.

**Comment by mkaz** [comment]
@timmyc the Attachment link is weird - it works for some of my test images but not others, trying to figure out why since the data should be accurate

**Comment by timmyc** [comment]
@mkaz so I thought it might have just been certain images in my library too... but I tested with the same images using the gallery flow in the current editor and the links worked okay for me there with "attachment" selected.

I didn't debug further, but thought something in the following line might be the culprit:

`if ( props.linkto === 'attachment' && !! props.img.link ) {`

File: blocks/library/gallery/gallery-image.js [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+		return (
+			<figure className="blocks-gallery-image">
+				<a href={ props.img.url }>
+					<img src={ props.img.url } alt={ props.img.alt } />
+				</a>
+			</figure>
+		);
+	}
+
+	if ( props.linkto === 'attachment' && !! props.img.link ) {
```
Comment (mtias) [inline blocks/library/gallery/gallery-image.js]: It'd be nice to not duplicate the markup for links and do this instead:
https://github.com/WordPress/gutenberg/blob/master/blocks/library/image/index.js#L200

Comment (mkaz) [inline blocks/library/gallery/gallery-image.js]: :+1:

File: blocks/library/gallery/index.js [hunk 3]
```diff
@@ -71,9 +77,11 @@ registerBlockType( 'core/gallery', {
 
 	edit( { attributes, setAttributes, focus, className } ) {
 		const { images = [], columns = defaultColumnsNumber( attributes ), align = 'none' } = attributes;
+		const setLinkto = ( value ) => setAttributes( { linkto: value } );
```
Comment (mtias) [inline blocks/library/gallery/index.js]: tiny thing, but probably should camel case the `linkTo` :)

File: blocks/library/gallery/index.js [hunk 4]
```diff
@@ -71,9 +77,11 @@ registerBlockType( 'core/gallery', {
...
 		const setColumnsNumber = ( event ) => setAttributes( { columns: event.target.value } );
 		const updateAlignment = ( nextAlign ) => setAttributes( { align: nextAlign } );
 		const { imageCrop = true } = attributes;
+		const { linkto = 'none' } = attributes;
```
Comment (mtias) [inline blocks/library/gallery/index.js]: These could be moved to default attrs:
```
	defaultAttributes: {
		dropCap: false,
	},
```

**Comment by jasmussen** [comment]
Also one niggle: I'd move "Link To" down below the "Crop Images" box:

screen shot 2017-08-04 at 19 32 12

**Comment by mtias** [comment]
Also maybe the label + input could all fit in one row?

**Comment by mkaz** [comment]
@mtias I'm just using the SelectControl component and don't see any real options on how to keep it all on the same row - any suggestion?

**Comment by mtias** [comment]
That's fine, we could look into the component itself in the future.

File: blocks/library/gallery/gallery-image.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ export default function GalleryImage( props ) {
 		case 'media':
 			href = props.img.url;
 			break;
-		case 'attacment':
+		case 'attachment':
```
Comment (timmyc) [inline blocks/library/gallery/gallery-image.js]: i always misspell that word too - funny that i didn't catch that when reviewing. 😄

**Pull request merged**